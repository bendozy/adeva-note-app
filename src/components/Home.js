import React from 'react';
import Grid from '@material-ui/core/Grid';
import Typography from '@material-ui/core/Typography';
import { makeStyles } from '@material-ui/core/styles';

const useStyles = makeStyles(theme => ({
  body: {
    marginTop: theme.spacing(5),
  },
}));

const Home = () => {
  const classes = useStyles();

  return (
    <Grid>
      <Grid>
        <Typography variant="h4" color="inherit" noWrap>
          Welcome
        </Typography>
      </Grid>
      <Grid>
        <Typography
          variant="body1"
          component="p"
          color="inherit"
          className={classes.body}
        >
          Hello there, thanks you for checking out this simple notes application.
          With this app, you can create, update, delete and search for notes.
        </Typography>
        <Typography
          variant="body1"
          component="p"
          color="inherit"
          className={classes.body}
        >
          This appliction uses localStorage for persistence. To get startted,
          click on the &quot;Add New Note&quot; button on the sidebar.
        </Typography>
      </Grid>
    </Grid>
  );
};

export default Home;
