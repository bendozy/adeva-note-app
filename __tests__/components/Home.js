import React from 'react';
import toJson from 'enzyme-to-json';
import Typography from '@material-ui/core/Typography';
import Home from '../../src/components/Home';

describe('<Home />', () => {
  it('should render a <Home /> component with Typography components', () => {
    const wrapper = shallow(<Home />);

    expect(wrapper.find(Typography)).toHaveLength(3);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
