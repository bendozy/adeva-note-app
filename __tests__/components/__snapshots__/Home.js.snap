// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> should render a <Home /> component with Typography components 1`] = `
<WithStyles(ForwardRef(Grid))>
  <WithStyles(ForwardRef(Grid))>
    <WithStyles(ForwardRef(Typography))
      color="inherit"
      noWrap={true}
      variant="h4"
    >
      Welcome
    </WithStyles(ForwardRef(Typography))>
  </WithStyles(ForwardRef(Grid))>
  <WithStyles(ForwardRef(Grid))>
    <WithStyles(ForwardRef(Typography))
      className="makeStyles-body-1"
      color="inherit"
      component="p"
      variant="body1"
    >
      Hello there, thanks you for checking out this simple notes application. With this app, you can create, update, delete and search for notes.
    </WithStyles(ForwardRef(Typography))>
    <WithStyles(ForwardRef(Typography))
      className="makeStyles-body-1"
      color="inherit"
      component="p"
      variant="body1"
    >
      This appliction uses localStorage for persistence. To get startted, click on the "Add New Note" button on the sidebar.
    </WithStyles(ForwardRef(Typography))>
  </WithStyles(ForwardRef(Grid))>
</WithStyles(ForwardRef(Grid))>
`;
